### PR TITLE
feat(profiling): add some helper classes

### PR DESCRIPTION
**Pull request by JonasBa**
I'm starting to move over the profiling import code. The following are just a few classes we use to construct profiles and a small utility to resolve stacks. In the next few PR's, I'll start moving the actual code importing code.

### CODE REVIEW

**Comment by JonasBa** [comment]
I am def not used to stacking PR's on GH

File: static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export function stackMarkerToHumanReadable(marker: JSSelfProfiling.Marker): string {
+  switch (marker) {
+    case 'gc':
+      return 'Garbage Collection';
```
Comment (matejminar) [inline static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts:4]: If this gets displayed we should make it translatable:
```suggestion
      return t('Garbage Collection');
```
Same for all other strings in this PR.

t comes from `import {t} from 'sentry/locale';`

**Review by matejminar — COMMENTED** [review]
Generally looks good, left a comment about translations.

One more thing to consider is that we use `.tsx` and `.spec.tsx` extensions everywhere. Even if the file does not contain jsx.

**Comment by JonasBa** [comment]
@matejminar thanks for the review! I think repo navigation and discoverability in general could benefit from using .ts instead of .tsx when files dont contain any jsx. If you consider the case where we might have a Frame data structure and a Frame.tsx which will be responsible for rendering the Frame component, then naming both files .tsx will lose that notion

File: static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      return t('Layout');
+    case 'paint':
+      return t('Paint');
+    case 'script':
+      return t('Script');
+    case 'other':
+      return t('Other');
+    default:
+      // since spec is still in dev, just gracefully return whatever we received.
+      return t(marker);
```
Comment (matejminar) [inline static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts:19]:
```suggestion
      return marker;
```

**Review by matejminar — APPROVED** [review]
I'm fine with merging as is, but please do bring up the topic of `ts` vs `tsx` in TSC so that we make a team decision.

**Comment by JonasBa** [comment]
@matejminar I'm going to update those files too, idk why I thought that convention only applies to tests 🤦🏼

File: static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts (hunk 2 above)
Comment (Zylphrex) [reply static/app/utils/profiling/formatters/stackMarkerToHumanReadable.ts:19]: This doesn't actually work as `t(...)` only works on literals, and not variables.

**Comment by JonasBa** [comment]
@matejminar @Zylphrex any ideas why the linting is failing for getsentry? I'll try rebase to master, see if it was fixed upstream

**Comment by Zylphrex** [comment]
> @matejminar @Zylphrex any ideas why the linting is failing for getsentry? I'll try rebase to master, see if it was fixed upstream

Looks like you just need to rebase on master

**Pull request merged**